src/Data/OKRDataService.ts

```diff
@@ -3,7 +3,7 @@ import * as SDK from "azure-devops-extension-sdk";
 import { CommonServiceIds, IExtensionDataService, IExtensionDataManager, IProjectPageService } from "azure-devops-extension-api";
 
 export abstract class OKRDataService<T extends OKRDocument> {
-    protected abstract getDataCollectionKey(): string;
+    protected abstract getDataCollectionKey(additionalKey?: string): string;
```

Comment:
I add this additional key prop to our data service. The other options is modifying the ObjectiveService, but it felt a little sloppier. Thoughts?

Reply:
I like the idea. But I'll still need to pass in what suffix I want to use to the service somehow. The time frame id won't be static for the lifetime of the ObjectiveService. I can either update within the objective service every time my timeframe changes or will need to continue to pass it in to each individual method. Let's chat on Monday, I think your proposal is better but I'd like to chat about implementation.